src/server/client_handler.c

```diff
@@ -6,48 +6,47 @@
 #include <sys/socket.h>
 #include <unistd.h>
 
-void* handle_client(void* arg) {
-  // Detach the current thread.
-  pthread_detach(pthread_self());
-
-  client_handler_args_t* handler_args = (client_handler_args_t*)arg;
-
-  int* client_sockets = handler_args->client_sockets;
-  int sock = handler_args->sock;
-  int* n_clients = handler_args->n_clients;
-  pthread_mutex_t* clients_mutex = handler_args->clients_mutex;
+#include "../util/util.h"
 
-  char buffer[BUFFER_SIZE];
-  ssize_t read_size = 0;
+void assign_handler(client_handler_args_t* handler_args) {
+  pthread_t thread_id = 0;
+  if (pthread_create(&thread_id, NULL, handle_client, (void*)handler_args) !=
+      0) {
+    error_and_exit("Error creating thread");
```

Comment:
Is the `n_clients` from the `handler_args` being freed somewhere before it exits?